packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -50,11 +61,16 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
     with AutoDisposeMixin {
   TimelineController controller;
 
+  TimelineMode get timelineMode => controller.timelineModeNotifier.value;
+
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     controller = Controllers.of(context).timeline;
     controller.timelineService.updateListeningState(true);
+
+    cancel();
+    addAutoDisposeListener(controller.timelineModeNotifier, refresh);
```

Comment:
suggestion: we can make refresh private, and make the second parameter to this method an optional positional argument.  When not provided, default to refresh.